sse/sse.go

```diff
@@ -209,61 +290,71 @@ func (es *EventSource) dispatch() {
             ch, exists := es.channels[c.channel]
 
             if !exists {
-                ch = make(map[*Client]bool)
+                ch = &Channel{
+                                new(sync.Mutex),
+                                "",
+                                make(map[*Client]bool),
+                                0,
+                            }
                 es.channels[c.channel] = ch
                 log.Info("New channel '%s' created.", c.channel)
             }
 
-            ch[c] = true
+            ch.UpdateExpiration()
+
+            ch.clientsConnected[c] = true
             log.Info("Client '%s' connected to channel '%s'.", c.ip, c.channel)
+            lastMessage := ch.GetLastMessage()
+
+            if (es.UseLastMessage && (len(lastMessage) > 0)) {
```

Comment:
Opss! You only can send last event id if you got the header `Last-Event-ID` and it's a value lower than the current id. This is how specs works.